third_party/move/tools/move-unit-test/src/test_reporter.rs

```diff
@@ -26,6 +29,62 @@ use std::{
     time::Duration,
 };
 
+/// A gas schedule where every instruction has a cost of "1". This is used to bound execution of a
+/// test to a certain number of ticks.
+fn unit_cost_table() -> CostTable {
+    let mut cost_schedule = zero_cost_schedule();
+    cost_schedule.instruction_table.iter_mut().for_each(|cost| {
+        *cost = GasCost::new(1, 1);
+    });
+    cost_schedule
+}
+
+pub trait UnitTestFactory<G: GasMeter> {
+    fn new_gas_meter(&self) -> G;
+    fn finish_session(
+        &self,
+        session: Session,
+        gas_meter: G,
+        test_run_info: TestRunInfo,
+    ) -> (VMResult<ChangeSet>, TestRunInfo);
```

Comment:
I feel like we are going a bit too far by letting the Factory taking over `finish_session`. 

It would be a better design to replace this with a more restricted `fn finalize_test_run_info(gas_meter: G, test_run_info: TestRunInfo) -> TestRunInfo`.

Now thinking about the future, I think we might need even more customization to support client needs, namely the ability to define custom run info and render them. We can worry about this later though -- the factory trait already lays down the ground work to make this easy.